File: back/src/main/java/com/back/domain/trend/snapshot/entity/DailyMessageCount.java
package com.back.domain.trend.snapshot.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Getter
@NoArgsConstructor
@Table(name = "daily_message_count",
        uniqueConstraints = @UniqueConstraint(columnNames ={"date"}))
public class DailyMessageCount {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDate date;
    private long totalMessages;

    public DailyMessageCount(LocalDate date, long totalMessages) {
        this.date = date;
        this.totalMessages = totalMessages;
    }

    public void updateTotalMessages(long totalMessages) {
        this.totalMessages = totalMessages;
    }
}

File: back/src/main/kotlin/com/back/domain/trend/snapshot/entity/DailyMessageCount.kt
package com.back.domain.trend.snapshot.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import java.time.LocalDate

@Entity
@Table(
    name = "daily_message_count",
    uniqueConstraints = [UniqueConstraint(columnNames = ["date"])]
)
class DailyMessageCount(
    date: LocalDate,
    totalMessages: Long
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
        protected set

    var date: LocalDate = date
        protected set

    var totalMessages: Long = totalMessages
        protected set

    fun updateTotalMessages(totalMessages: Long) {
        this.totalMessages = totalMessages
    }
}
